decorators/uncontrolled.jsx

```diff
@@ -35,7 +35,13 @@ export default ({defaultProp, prop, handlerName, handlerSelector, resetHandlerNa
 
   render () {
     const props = {
-      ...this.props,
+      ...Object.keys(this.props).reduce((filtered, propName) => {
```

Comment:
Consider breaking this up into an explicit filter step and an explicit copy step to make it clearer.